app/src/org/commcare/android/analytics/XPathErrorEntry.java

```diff
@@ -0,0 +1,130 @@
+ */
+public class XPathErrorEntry extends LogEntry implements Persistable, IMetaData {
+    public static final String STORAGE_KEY = "XPATH_ERROR";
+    private static final String TAG = XPathErrorEntry.class.getSimpleName();
+    private static final String META_DATE = "date";
+
+    private int recordId = -1;
+    private int appVersion;
+    private String expression;
+    private String sessionFramePath;
+
+    public XPathErrorEntry() {
+        // for externalization
+    }
+
+    protected XPathErrorEntry(String expression, String errorMessage) {
+        super(AndroidLogger.TYPE_ERROR_CONFIG_STRUCTURE, errorMessage, new Date());
+
+        this.expression = expression;
+        this.sessionFramePath = getCurrentSession();
+        this.appVersion = getAppVersion();
+    }
+
+    private static String getCurrentSession() {
+        CommCareSession currentSession;
+        try {
+            currentSession = CommCareApplication._().getCurrentSession();
+            // TODO PLM:
+            // return currentSession.getFrame().sessionPathString();
+            return "";
+        } catch (SessionStateUninitException e) {
+            return "";
+        }
+    }
+
+    private int getAppVersion() {
+        CommCareApp app = CommCareApplication._().getCurrentApp();
+
+        if (app != null) {
+            return app.getCommCarePlatform().getCurrentProfile().getVersion();
+        }
+
+        return -1;
+    }
+
+    public String getExpression() {
+        return expression;
+    }
+
+    public String getSessionPath() {
+        return sessionFramePath;
+    }
+
+    @Override
+    public String toString() {
+        return getTime().toString() + " | " +
+                getMessage() + " caused by " + expression +
+                "\n" + "session: " + sessionFramePath;
+    }
+
+    @Override
+    public void setID(int ID) {
+        recordId = ID;
+    }
+
+    @Override
+    public int getID() {
+        return recordId;
+    }
+
+    @Override
+    public void readExternal(DataInputStream in, PrototypeFactory pf)
+            throws IOException, DeserializationException {
+        super.readExternal(in, pf);
+
+        recordId = ExtUtil.readInt(in);
+        appVersion = ExtUtil.readInt(in);
+        expression = ExtUtil.readString(in);
+        sessionFramePath = ExtUtil.readString(in);
+    }
+
+    @Override
+    public void writeExternal(DataOutputStream out) throws IOException {
+        super.writeExternal(out);
+
+        ExtUtil.writeNumeric(out, recordId);
+        ExtUtil.writeNumeric(out, appVersion);
+        ExtUtil.writeString(out, expression);
+        ExtUtil.writeString(out, sessionFramePath);
+    }
+
+    @Override
+    public String[] getMetaDataFields() {
+        return new String[]{META_DATE};
+    }
+
+    @Override
+    public Object getMetaData(String fieldName) {
+        if (META_DATE.equals(fieldName)) {
+            return DateUtils.formatDate(getTime(), DateUtils.FORMAT_ISO8601);
+        }
```

Comment:
Unsure what meta data I should be exposing in the db entries for these error logs.